### ISSUE TITLE
Confirmation on sharing state between multiple instances of Eclipser

### ISSUE THREAD

**Issue by arizvisa** [SoftSec-KAIST/Eclipser#12]
I'd like to run multiple instances of Eclipser, is it just as performant to export the entire `output` directory? Or can I prioritize it so that `/.coverage` and `/.internal` (which I think contains the seed queues?) are on a fast-backing, and `/crash`, and `/testcase` are on slower disk-type backing?

**Comment by jchoi2022** [comment]
I apologize for the late reply. I accidentally missed the alarm.

First, `$OUTDIR/crash` contains bug-triggering inputs, and `$OUTDIR/testcase` contains inputs that increased node coverage. Thus these are the most important inputs to care about. For example, you can measure the node/edge coverage achieved by Eclipser, by replaying the seeds in `$OUTDIR/testcase`.

Next, `$OUTDIR/.internal` some more clarification. Eclipser internally manages (1) high-priority seed queues, and (2) low-prioirty seed queues. A seed is assigned high-priority if it increases node coverage, and assigned low-priority if it only increases path coverage. High priority seeds are immediately stored in `$OUTDIR/testcase` as final output, and also cached in memory for fast access. On the other hand, low priority queue is relatively large, so stored in file system, at `$OUTDIR/.internal`. Other files like `$OUTDIR/.coverage` can be simply ignored, as they are temporary files internally used by Eclipser.

To summarize, it would be better to prioritize seeds in `$OUTDIR/testcase` directory, as they are more important seeds.

By the way, be cautious that seeds in `$OUTDIR/testcase` are in JSON format, not raw input. So if you are planning something like the seed queue sharing between multiple instances, it can be problematic and may need some code change. If you are simply summing up the final results (i.e. crashes and test cases) of multiple Eclipser instances, then it's fine.

Again, I apologize for the late reply, and feel free to ask any other question if you have one.

**Comment by arizvisa** [comment]
Ok. I was intending to do exactly that (sharing the seed queue between multiple instances). I updated the code to expose those different paths to the commandline, but haven't tested it yet. I'll probably end up re-working some of that stuff. Thanks!

**Comment by arizvisa** [comment]
Closing this, as my question as answered.
